src/backend/wiki_game_info/wiki_game_info.ts

```diff
@@ -0,0 +1,67 @@
+import { wikiGameInfoStore } from './electronStore'
+import { removeSpecialcharacters } from '../utils'
+import { WikiInfo } from '../../common/types'
+import { logError, logInfo, LogPrefix } from '../logger/logger'
+import { getInfoFromAppleGamingWiki } from './applegamingwiki/utils'
+import { getHowLongToBeat } from './howlongtobeat/utils'
+import { getInfoFromPCGamingWiki } from './pcgamingwiki/utils'
+import { isMac } from '../constants'
+
+export async function getWikiGameInfo(
+  title: string,
+  gogID?: string
+): Promise<WikiInfo | null> {
+  try {
+    title = removeSpecialcharacters(title)
+
+    // check if we have a cached response
+    const cachedResponse = wikiGameInfoStore.get(title) as WikiInfo
+    if (cachedResponse) {
+      logInfo(
+        [`Using cached ExtraGameInfo data for ${title}`],
+        LogPrefix.ExtraGameInfo
+      )
+
+      const oneMonthAgo = new Date()
+      oneMonthAgo.setMonth(oneMonthAgo.getMonth() - 1)
+
+      const timestampLastFetch = new Date(cachedResponse.timestampLastFetch)
+      if (timestampLastFetch > oneMonthAgo) {
+        return cachedResponse
+      }
+
+      logInfo(
+        [`Cached ExtraGameInfo data for ${title} outdated.`],
+        LogPrefix.ExtraGameInfo
+      )
+    }
+
+    logInfo(`Getting ExtraGameInfo data for ${title}`, LogPrefix.ExtraGameInfo)
+
+    const pcgamingwiki = await getInfoFromPCGamingWiki(title, gogID)
+    const applegamingwiki = isMac
+      ? await getInfoFromAppleGamingWiki(title)
+      : null
+    const howlongtobeat = await getHowLongToBeat(
+      title,
+      pcgamingwiki?.howLongToBeatID ?? ''
+    )
```

Comment:
maybe we can do this 3 requests in parallel instead of sequentially?

then we can await on a Promise.all

Reply:
also, this is wrapped with a try/catch, if any of the requests fail this all fails? for example if for some reason howlongtobeat is down this will make the pcgamingwiki fail

Reply:
Should not happen because all three methods are also wrapped in try/catch. The try/catch here is only for electron-store. Don't know if that can throw something.
If not we could remove the try/catch.